exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/MysqlDialect.kt

```diff
@@ -362,13 +362,18 @@ open class MysqlDialect : VendorDialect(dialectName, MysqlDataTypeProvider, Mysq
     /** Returns `true` if a MySQL JDBC connector is being used and its version is greater than or equal to 8.0. */
     fun isTimeZoneOffsetSupported(): Boolean = (currentDialect !is MariaDBDialect) && isMysql8
 
+    private val notAcceptableDefaults = mutableListOf("CURRENT_DATE()", "CURRENT_DATE")
+
     override fun isAllowedAsColumnDefault(e: Expression<*>): Boolean {
         if (super.isAllowedAsColumnDefault(e)) return true
         if ((currentDialect is MariaDBDialect && fullVersion >= "10.2.1") || (currentDialect !is MariaDBDialect && fullVersion >= "8.0.13")) {
             return true
         }
-        val acceptableDefaults = mutableListOf("CURRENT_TIMESTAMP", "CURRENT_TIMESTAMP()", "NOW()", "CURRENT_TIMESTAMP(6)", "NOW(6)")
-        return e.toString().trim() in acceptableDefaults && isFractionDateTimeSupported()
+
+        // This check is quite optimistic, it will not allow to create a varchar columns with "CURRENT_DATE" default value for example
+        // Comparing to the previous variant with white list of functions the new variant does not reject valid values,
+        // it could be checked on the test UpsertTests::testUpsertWithColumnExpressions()
+        return e.toString().trim() !in notAcceptableDefaults
```

Comment:
The main idea of the change that before that change any value that is not in the list (event if it's valid default value) was recognized as invalid default value, and skipped in DDL.

Actually, I can say I see straightforward solution here. At the current moment we have some features to help user if he tries to put non valid default value. That includes:

- prevent creating DDL with invalid value
- convert column into nullable one if there is no client default value
- sending default value in every insert query

My PR removes the third case.

One option is to try to continue what we did before, and for example add not only conversion to nullable column, but also conversion `default` column into `clientDefaultExpression` column, or we can throw errors and ask users not to put invalid values into default columns (kind of removing the magic we do for users now), or probably there are other variants I don't see now.

@joc-a do you remember why we added such a logic? Probably you also have some thoughts what better to do if we loose sending all the defaults in inserts.

Reply:
According to `clientDefaultExpression()` as public API, we had already some discussions on it, but those time it was added more for feature completeness (if we have `default`, `clientDefault`, and `defaultExpression`, why we should not add `clientDefaultExpresson`), but decided that's probably not the thing that is needed by many people.
But with new reality it could be a good idea to add it. I could add it with refactoring of current 3 fields for default values.